common/server.cpp

```diff
@@ -168,6 +170,15 @@ AuthenticationResult Server::loginUser(Server_ProtocolHandler *session, QString
     event.mutable_user_info()->CopyFrom(session->copyUserInfo(true, true, true));
     locker.unlock();
     
+    // check if client id exists (older client compatibility)
+    if (clientid.isEmpty()){
+        // client id is empty, either out dated client or client has been modified
+    }
+    else {
+        // update users database table with client id
+        databaseInterface->updateUsersClientID(name, clientid);
```

Comment:
One line is fine.

And for updates you mentioned locks, I was going off that

On Fri, Aug 7, 2015, 10:11 AM woogerboy21 <notifications@github.com> wrote:

> In common/server.cpp
> <https://github.com/Cockatrice/Cockatrice/pull/1340#discussion_r36522183>:
>
> > @@ -168,6 +170,15 @@ AuthenticationResult Server::loginUser(Server_ProtocolHandler *session, QString
> >      event.mutable_user_info()->CopyFrom(session->copyUserInfo(true, true, true));
> >      locker.unlock();
> >
> > +    // check if client id exists (older client compatibility)
> > +    if (clientid.isEmpty()){
> > +        // client id is empty, either out dated client or client has been modified
> > +    }
> > +    else {
> > +        // update users database table with client id
> > +        databaseInterface->updateUsersClientID(name, clientid);
>
> There is no lock being made when updating the client id on the user table?
> No multiple locks are going on here. Now you could say that maybe we should
> apply the client id inside the locking block but I thought you wanted to
> move away from locking the user table?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Cockatrice/Cockatrice/pull/1340/files#r36522183>.
>